### DISCUSSION TITLE
Use an existing ssh connection for the NetconfDriver / AsyncNetconfDriver

### DISCUSSION THREAD

**Discussion by AndryNick98** [scrapli/scrapli_netconf#141] [Q&A]
Hi everyone,
I'd like to know if with scrapli_netconf i'm able to using an existing ssh connection (created before, using for example asyncssh) to execute netconf operations,  so without establish a new ssh connection.
For example, suppose that i have already established a ssh connection to execute some commands and i want to use same connection to perform even netconf operations.
Does scrapli_netconf provide any support for that ?

**Comment by carlmontanari** [comment]
directly, "no", and probably wouldn't -- mostly because I dont think it would be easy to consistently provide that behavior across all transport flavors.

is it possible? maybe? probably? if asyncssh can do it, then im sure you could patch scrapli to make it happen - maybe by patching `open_netconf` in asyncssh transport plugin to basically not open anything but instead to just open the new channel and enable the sub system.
